astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -130,9 +136,11 @@ def icrs_to_gcrs(icrs_coo, gcrs_frame):
         # BCRS coordinate direction and *then* run the ERFA transform for no
         # parallax/PM. This ensures reversiblity and is more sensible for
         # inside solar system objects
-        newxyz = icrs_coo.cartesian.xyz.T - astrom['eb']*u.au
-        newcart = CartesianRepresentation(newxyz.T)
-
+        newxyz = intermedrep.to_cartesian().xyz
+        newxyz = np.rollaxis(newxyz,-1,0) - astrom['eb']*u.au
```

Comment:
And another time (sorry that my multi-D coordinates are not working yet; otherwise a simple test would have found this...).